backend/certificate-pdf-service/main.go

```diff
@@ -128,9 +128,119 @@ func getOtherOrganCertificateType(otherOrgans string) string {
+	pdf.AddPage()
+	if err := pdf.AddTTFFont("dev", "NotoSansDevanagari.ttf"); err != nil {
+		log.Printf(err.Error())
+		return nil, err
+	}
+	otherOrganCertificateType := getOtherOrganCertificateType(certificateData.CredentialSubject.Pledge.AdditionalOrgans)
+	certificateUrl := templateUrl + otherOrganCertificateType
+	fileName := strings.Split(certificateUrl, "/")[len(strings.Split(certificateUrl, "/"))-1]
+	err := getTemplateFile(certificateUrl, fileName)
+	if err != nil {
+		log.Printf("Error in certificate download : %v", err)
+		return nil, err
+	}
+	tpl1 := pdf.ImportPage(fileName, 1, "/MediaBox")
+	pdf.UseImportedTemplate(tpl1, 0, 0, 0, 0)
+	if err := pdf.SetFont("dev", "", 18); err != nil {
+		log.Print(err.Error())
+		return nil, err
+	}
+	offsetX := 163.0
+	offsetY := 160.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.Name)
+
+	if err := pdf.SetFont("dev", "", 10); err != nil {
+		log.Print(err.Error())
+		return nil, err
+	}
+	offsetX = 164.0
+	offsetY = 291.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	date, err := time.Parse(time.RFC3339, certificateData.IssuanceDate)
+	if err != nil {
+		log.Printf("Error : %v", err.Error())
+		return nil, err
+	}
+	_ = pdf.Cell(nil, date.Format("02-03-2006"))
+
+	offsetX = 164.0
+	offsetY = 327.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.Evidence[0].RefId)
+
+	offsetX = 164.0
+	offsetY = 361.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.NottoId)
+
+	offsetX = 164.0
+	offsetY = 395.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.Pledge.Organs)
+
+	offsetX = 164.0
+	offsetY = 429.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.Pledge.Tissues)
+
+	offsetX = 464.0
+	offsetY = 291.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.FatherName)
+
+	offsetX = 464.0
+	offsetY = 327.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.BloodGroup)
+
+	offsetX = 464.0
+	offsetY = 361.0
+	pdf.SetX(offsetX)
+	pdf.SetY(offsetY)
+	_ = pdf.Cell(nil, certificateData.CredentialSubject.Emergency.MobileNumber)
+
+	holder, err := gopdf.ImageHolderByBytes(qrData)
+	if err = pdf.ImageByHolder(holder, 544, 276, &gopdf.Rect{W: 210, H: 210}); err != nil {
+		log.Errorf("Error creating QR Code")
+	}
+	photoStr, err := base64.StdEncoding.DecodeString(string(photo))
+	if err != nil {
+		return nil, err
+	}
+	holder, err = gopdf.ImageHolderByBytes(photoStr)
+	if err = pdf.ImageByHolder(holder, 50, 161, &gopdf.Rect{W: 110, H: 110}); err != nil {
+		log.Errorf("Error creating Profile photo")
+	}
+
+	var b bytes.Buffer
+	_ = pdf.Write(&b)
+	return b.Bytes(), nil
+}
+
+func renderPortraitPdf(pdf gopdf.GoPdf, templateUrl string, certificateData Certificate, qrData []byte, photo []byte) ([]byte, error) {
```

Comment:
can we create a generic function for both portrait and landscape, which takes map of value and its positions, and the template